rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,133 @@
+
+
+## What should a location tell you?
+
+When somebody looks at an item, they want to know two things about the location:
+
+*   **Am I allowed to see it?**
+    This is encoded by the *access conditions* field, which includes an *access status* (drawn from a fixed vocabulary: open, closed, requires-permission, etc.) and an *access description* (free-text, written by cataloguers).
+
+*   **If I'm allowed to see it, where can I find it?**
+    For example: where it sits on the physical shelves within the library, or a particular URL, or a IIIF resource.
+
+We need to support filtering and aggregating by location.
+For example, users should be able to find all the material that's openly available online (if they're working remotely), or all the material in the library (if they're in the building).
+
+
+
+## What locations do we need to support?
+
+These are the types of location our model needs to support:
+
+*   **Open shelves.**
+
+    This should include a *shelf location* (e.g. History of Medicine Collection) and a *shelfmark* (e.g. TW.AL) so that a user can find it.
+
+*   **Closed stores.**
+
+    Although the MARC records include a shelf location and a shelfmark, the Collections team would prefer this not to be exposed publicly.
+
+    Do we need to expose this information for the requesting service?
+
+    -   Can something be requested faster if it's stored on-site?
+    -   Does the requesting service need to know the location of something when it tells Sierra "please ask somebody to fetch this item", or can it just provide Sierra the item ID?
+
+*   **A standalone IIIF image.**
+
+    For example, the Miro images.
+    This should include a *URL* (to the IIIF resource), and optionally a *license* and *credit line*.
+
+*   **A IIIF presentation manifest.**
+
+    For example, the digitised books.
+    As with a standalone IIIF image, this should include a *URL*, and optionally a *license* and *credit line*.
+
+*   **External websites or URLs.**
+
+    For example, a journal with online access.
+    This should include a *URL* (to access the resource), and optional *link text* (e.g. View this e-book).
+
+    If it's a URL with subscription access, users can only view the journal by accessing a URL like `https://{proxyname}.catalogue.wellcomelibrary.org` and logging in to their Library account.
+    This means we need an access status of `login-required`, although this isn't explicitly communicated in the holdings records -- we'd have to infer it from the URL.
+
+*   **On exhibition.**
+
+    This should include the *name of the exhibition* and the *due date* when the item will be returning to the library.
+
+    For an example of this, see <https://wellcomecollection.org/works/azrhju85>.
+    There might be a free-form text description that lives separately in the "notes" field.
+
+There are a number of other infrequently used location codes (e.g. "offsite", "at digitisation", "Rare Materials Room") that we currently present in the API.
```

Comment:
Agree it's not worth parsing specific locations and dates out.

What we do currently is look in the "location" field on the item record, which returns `exres` for "on-exhibition", then present the free text description as an ExhibitionNote from MARC 585.

Reply:
Ah ok, we should do that then and make exres one of the exceptional five letter location codes that we map explicitly.

Reply:
That's true for 'at digitisation' but @alexwlchan is correct that `hgser` (offsite) and `wghip` / `wqrfp` (rare materials room - don't ask me why there are two) are locations, not OPACMSG. The use of OPACMSG in combination with these is not consistent...

Reply:
For the catalogue API I'm going to make this "Closed stores", because it's quite a bit easier, and we can return to the "how do you request this thing" question for the requesting service.

Particularly if we want to channel all of these things through online requesting, offsite or not, let's not perpetuate the distinction right now.